runtime/src/accounts_background_service.rs

```diff
@@ -155,12 +156,16 @@ impl SnapshotRequestHandler {
                 };
                 hash_time.stop();
 
+                let last_full_snapshot_slot = snapshot_utils::calculate_last_full_snapshot_slot(
+                    snapshot_root_bank.slot(),
+                    self.snapshot_config.full_snapshot_archive_interval_slots,
+                );
                 let mut clean_time = Measure::start("clean_time");
                 // Don't clean the slot we're snapshotting because it may have zero-lamport
                 // accounts that were included in the bank delta hash when the bank was frozen,
                 // and if we clean them here, the newly created snapshot's hash may not match
                 // the frozen hash.
-                snapshot_root_bank.clean_accounts(true, false, None);
+                snapshot_root_bank.clean_accounts(true, false, Some(last_full_snapshot_slot));
```

Comment:
Calculate the last full snapshot slot and use it when calling `clean_accounts()`. This happens again later in the file too.

Reply:
You could track the `last_full_snapshot_slot` every time this condition is met: https://github.com/solana-labs/solana/blob/6d246031d06b81ee21d591a267a50f9f76d93eac/runtime/src/accounts_background_service.rs#L178-L182, and just pass it in here.

On startup you infer this `last_full_snapshot_slot` from the snapshot you unpack

Reply:
Good idea, will do.

Reply:
Done!